hudi-spark-datasource/hudi-spark/src/main/java/org/apache/hudi/QuickstartUtils.java

```diff
@@ -121,11 +121,23 @@ public static GenericRecord generateGenericRecord(String rowKey, String riderNam
      */
     public static OverwriteWithLatestAvroPayload generateRandomValue(HoodieKey key, String riderDriverSuffix)
         throws IOException {
+      // The timestamp generated is limited to range from 7 days before to now, to avoid generating too many
+      // partitionPaths when user use timestamp as partitionPath filed.
       GenericRecord rec =
-          generateGenericRecord(key.getRecordKey(), "rider-" + riderDriverSuffix, "driver-" + riderDriverSuffix, 0);
+          generateGenericRecord(key.getRecordKey(), "rider-" + riderDriverSuffix,"driver-"
+              + riderDriverSuffix, generateRangeRandomTimestamp(7));
       return new OverwriteWithLatestAvroPayload(Option.of(rec));
     }
 
+    /**
+     * Generate timestamp range from {@param daysTillNow} before to now.
+     */
+    private static long generateRangeRandomTimestamp(int daysTillNow) {
+      long now = System.currentTimeMillis();
+      long daysBefore = now - daysTillNow * 24 * 60 * 60 * 1000L;
+      return daysBefore + (long)(Math.random() * (now - daysBefore));
```

Comment:
It's not good for reading.  `now - daysBefore ` = `now - (now - daysTillNow * 24 * 60 * 60 * 1000L)`?

Reply:
> It's not good for reading. `now - daysBefore ` = `now - (now - daysTillNow * 24 * 60 * 60 * 1000L)`?

how about this
```
    private static long generateRangeRandomTimestamp(int daysTillNow) {
      long interval = daysTillNow * 24 * 60 * 60 * 1000L;
      return System.currentTimeMillis() - (long)(Math.random() * interval);
    }
```